README.md

```diff
@@ -1,76 +1,126 @@
+## How to run this sample
 
-3. In the left navigation sidebar, select **Azure Active Directory** service. Next, select **App registrations**.
+To run this sample, you'll need:
 
-5. Select **New application registration** and provide a friendly name for the app, app type, and sign-on URL:
-   **Name**: **Java-WebApp-OpenIDConnect**
-   **Application Type**: **Web app / API**
-   **Sign-on URL**: `http://localhost:8080/adal4jsample/`
-   Select **Create** to register the app.
+- Working installation of Java and Maven
+- Tomcat or any other J2EE container solution
+- An Internet connection
+- An Azure Active Directory (Azure AD) tenant. For more information on how to get an Azure AD tenant, see [How to get an Azure AD tenant](https://azure.microsoft.com/en-us/documentation/articles/active-directory-howto-tenant/)
+- A user account in your Azure AD tenant. This sample will not work with a Microsoft account (formerly Windows Live account). Therefore, if you signed in to the [Azure portal](https://portal.azure.com) with a Microsoft account and have never created a user account in your directory before, you need to do that now.
 
-6. Select **Settings** and on the **Keys** blade, enter a Key description, set a duration and click **Save**. Note down the key value that is displayed before closing the box since it will not be shown again.
+### Step 1: Download Java (8 and above) for your platform
 
-7. From the Azure portal, note the following information:
+To successfully use this sample, you need a working installation of [Java](http://www.oracle.com/technetwork/java/javase/downloads/index.html) and [Maven](https://maven.apache.org/).
 
-   **The Tenant domain:** See the **App ID URI** base URL under the **Properties** blade. For example: `contoso.onmicrosoft.com`
+### Step 2:  Clone or download this repository
 
-   The **Application ID** on the registered app blade. For example: `ba74781c2-53c2-442a-97c2-3d60re42f403`
+From your shell or command line:
 
-### Step 2: Download Java (7 and above) for your platform
+- `git clone https://github.com/Azure-Samples/active-directory-java-webapp-openidconnect.git`
 
-To successfully use this sample, you need a working installation of [Java](http://www.oracle.com/technetwork/java/javase/downloads/index.html) and [Maven](https://maven.apache.org).
+### Step 3:  Register the sample with your Azure Active Directory tenant
 
-### Step 3: Download the Sample application and modules
+To register these projects, you can:
 
-Next, clone the sample repo and install the project's dependencies.
+- either follow the steps in the paragraphs below 
+- or use PowerShell scripts that:
+  - **automatically** create for you the Azure AD applications and related objects (passwords, permissions, dependencies)
 
-From your shell or command line:
+If you want to use this automation, read the instructions in [App Creation Scripts](./AppCreationScripts/AppCreationScripts.md)
+
```

Comment:
Maybe you want to add a sentence "the configuration of your code (Step 4 below) still needs to be done manually"

Reply:
done